src/storage/IndyStorageService.ts

```diff
@@ -1,16 +1,15 @@
 import { inject, scoped, Lifecycle } from 'tsyringe'
 import type { WalletQuery, WalletRecord } from 'indy-sdk'
 
-import { StorageService } from './StorageService'
+import { StorageService, BaseRecordConstructor } from './StorageService'
 import { BaseRecord } from './BaseRecord'
 import { Wallet } from '../wallet/Wallet'
 import { JsonTransformer } from '../utils/JsonTransformer'
-import { Constructor } from '../utils/mixins'
 import { Symbols } from '../symbols'
 
-export interface BaseRecordConstructor<T> extends Constructor<T> {
-  type: string
-}
+import { handleIndyError, IndyError, isIndyError } from '../utils/indyError'
+import { AriesFrameworkError, RecordNotFoundError } from '../error'
+import { RecordDuplicateError } from '../error/RecordDuplicateError'
```

Comment:
Auto-import does some weird things sometimes. I've been fiddling a bit with eslint-plugin-import a bit so we can enforce (and auto-fix!) such things.

However they don't support the `import type` from TS very well so it will break your code